File: Code/backend/IndustrialGasAlarmCorporate/app/src/main/java/com/niit/industrialgasalarmcorporate/domain/content/Content.java
package com.niit.industrialgasalarmcorporate.domain.content;

import com.niit.industrialgasalarmcorporate.common.enums.ErrorCode;
import com.niit.industrialgasalarmcorporate.common.exception.BusinessException;

import java.time.LocalDateTime;
import java.util.UUID;

public class Content {

    private final String contentUuid;
    private String title;
    private String summary;
    private String body;
    private String coverImage;
    private ContentType type;
    private ContentStatus status;
    private String categoryUuid;
    private String categoryName;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    public Content(String title, String summary, String body, String coverImage,
                   ContentType type, ContentStatus status, String categoryUuid) {
        this.contentUuid = UUID.randomUUID().toString();
        this.title = title;
        this.summary = summary;
        this.body = body;
        this.coverImage = coverImage;
        this.type = type;
        this.status = status;
        this.categoryUuid = categoryUuid;
    }

    public Content(String contentUuid, String title, String summary, String body,
                   String coverImage, ContentType type, ContentStatus status, String categoryUuid,
                   String categoryName, LocalDateTime createdAt, LocalDateTime updatedAt) {
        this.contentUuid = contentUuid;
        this.title = title;
        this.summary = summary;
        this.body = body;
        this.coverImage = coverImage;
        this.type = type;
        this.status = status;
        this.categoryUuid = categoryUuid;
        this.categoryName = categoryName;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }

    public void publish() {
        if (this.status == ContentStatus.PUBLISHED) {
            throw new BusinessException(ErrorCode.VALIDATION_ERROR, "内容已是发布状态");
        }
        this.status = ContentStatus.PUBLISHED;
    }

    public void unpublish() {
        if (this.status != ContentStatus.PUBLISHED) {
            throw new BusinessException(ErrorCode.VALIDATION_ERROR, "仅已发布的内容可以取消发布");
        }
        this.status = ContentStatus.DRAFT;
    }

    public void update(String title, String summary, String body, String coverImage,
                       String categoryUuid) {
        this.title = title;
        this.summary = summary;
        this.body = body;
        this.coverImage = coverImage;
        this.categoryUuid = categoryUuid;
    }

    public String getContentUuid() {
        return contentUuid;
    }

    public String getTitle() {
        return title;
    }

    public String getSummary() {
        return summary;
    }

    public String getBody() {
        return body;
    }

    public String getCoverImage() {
        return coverImage;
    }

    public ContentType getType() {
        return type;
    }

    public ContentStatus getStatus() {
        return status;
    }

    public String getCategoryUuid() {
        return categoryUuid;
    }

    public String getCategoryName() {
        return categoryName;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }

    public void setStatus(ContentStatus status) {
        this.status = status;
    }
}

File: Code/frontend/src/types/content.ts
export enum ContentType {
  SOLUTION = 'SOLUTION',
  NEWS = 'NEWS'
}

export const ContentTypeMap: Record<ContentType, string> = {
  [ContentType.SOLUTION]: '解决方案',
  [ContentType.NEWS]: '新闻动态'
}

export enum ContentStatus {
  DRAFT = 'DRAFT',
  PUBLISHED = 'PUBLISHED'
}

export const ContentStatusMap: Record<ContentStatus, string> = {
  [ContentStatus.DRAFT]: '草稿',
  [ContentStatus.PUBLISHED]: '已发布'
}

export interface ContentVO {
  contentUuid: string
  title: string
  summary: string
  coverImage: string
  type: ContentType
  categoryUuid: string
  categoryName: string
  status: ContentStatus
  createdAt: string
  updatedAt?: string
}

export interface ContentDetailVO {
  contentUuid: string
  title: string
  body: string
  coverImage: string
  type: ContentType
  categoryUuid: string
  categoryName: string
  status: ContentStatus
  createdAt: string
  updatedAt: string
}

export interface CreateContentDTO {
  title: string
  type: ContentType
  categoryUuid: string
  body: string
  coverImage?: File
  status: ContentStatus
}

export interface UpdateContentDTO extends Partial<CreateContentDTO> {
  contentUuid: string
}
